AudioRecorder.py

```diff
@@ -38,22 +39,37 @@ def __init__(self):
 
 class DefaultSpeakerRecorder(BaseRecorder):
     def __init__(self):
-        with pyaudio.PyAudio() as p:
-            wasapi_info = p.get_host_api_info_by_type(pyaudio.paWASAPI)
+        #with pyaudio.PyAudio() as p:
+        p = pyaudio.PyAudio() 
+        try:
+            #wasapi_info = p.get_host_api_info_by_type(pyaudio.paWASAPI)
+            wasapi_info = p.get_host_api_info_by_type(pyaudio.paCoreAudio)
+
             default_speakers = p.get_device_info_by_index(wasapi_info["defaultOutputDevice"])
             
-            if not default_speakers["isLoopbackDevice"]:
-                for loopback in p.get_loopback_device_info_generator():
-                    if default_speakers["name"] in loopback["name"]:
-                        default_speakers = loopback
+            #if not default_speakers["isLoopbackDevice"]:
+            if not default_speakers.get("isLoopbackDevice", False):
+                
+                for i in range(p.get_device_count()):
+                    device_info = p.get_device_info_by_index(i)
+                    if device_info['maxInputChannels'] > 0 and device_info['hostApi'] == p.get_default_host_api_info()['index']:
+                        default_speakers = loopback = device_info
                         break
-                else:
-                    print("[ERROR] No loopback device found.")
-        
+                    else:
+                        print("[ERROR] No loopback device found.")
+
+                # for loopback in p.get_loopback_device_info_generator():
+                #     if default_speakers["name"] in loopback["name"]:
+                #         default_speakers = loopback
+                #         break
+                # else:
+                #     print("[ERROR] No loopback device found.")
+        finally:
+            p.terminate()
```

Comment:
This code doesn't make sense because `PyAudio` can't detect if the device is a loopback (`isLoopbackDevice` won't exist)

It would also be nice to change name of the "wasapi_*" variables, as they lose their meaning in this form.

Reply:
I did some changes in separate commits in order to support both Windows and macOS, I test it on my mac intel cpu and it worked perfectly through api "python main.py --api". I hope it will be approved and merged.
Thanks for this awesome repo.